.Jenkinsfile

```diff
@@ -0,0 +1,133 @@
+            }
+            options {
+                skipDefaultCheckout()
+            }
+            steps {
+                dir('build') {
+                    sh(
+                        label: 'Install/Update sources from Composer',
+                        script: 'COMPOSER_DISCARD_CHANGES=true composer update --prefer-source --no-interaction --no-ansi --no-progress --no-scripts'
+                    )
+                    sh(
+                        label: 'Add phpunit',
+                        script: 'composer require phpunit/phpunit:^8.5'
+                    )
+                    sh(
+                        label: "Extra filesystem mocks",
+                        script: '''
+mkdir -p taoQtiItem/views/js/mathjax/ && touch taoQtiItem/views/js/mathjax/MathJax.js
+mkdir -p tao/views/locales/en-US/
+    echo "{\\"serial\\":\\"${BUILD_ID}\\",\\"date\\":$(date +%s),\\"version\\":\\"3.3.0-${BUILD_NUMBER}\\",\\"translations\\":{}}" > tao/views/locales/en-US/messages.json
+mkdir -p tao/views/locales/en-US/
+                        '''
+                    )
+                }
+            }
+        }
+        stage('Tests') {
+            parallel {
+                stage('Backend Tests') {
+                    when {
+                        expression {
+                            fileExists('build/qtiItemPci/test/unit')
+                        }
+                    }
+                    agent {
+                        docker {
+                            image 'alexwijn/docker-git-php-composer'
+                            reuseNode true
+                        }
+                    }
+                    options {
+                        skipDefaultCheckout()
+                    }
+                    steps {
+                        dir('build'){
+                            sh(
+                                label: 'Run backend tests',
+                                script: './vendor/bin/phpunit qtiItemPci/test/unit'
+                            )
+                        }
+                    }
+                }
+                stage('Frontend Tests') {
+                    when {
+                        expression {
+                            fileExists('build/qtiItemPci/views/build/grunt/test.js')
+                        }
+                    }
+                    agent {
+                        docker {
+                            image 'btamas/puppeteer-git'
+                            reuseNode true
+                        }
+                    }
+                    environment {
+                        HOME = '.'
+                    }
+                    options {
+                        skipDefaultCheckout()
+                    }
+                    steps {
+                        dir('build/tao/views/build') {
+                            sh(
+                                label: 'Install tao-core frontend extensions',
+                                script: 'npm install'
```

Comment:
Something interesting: 

In the `tao-core` extension's `.JenkinsFile`, the same line is actually `script: 'npm install --production'`. We need to understand what is the difference and whether or not we need `--production`.

Reply:
It's like `no-dev` for composer. `--production` -> no devdepencies are installed.